src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/RethrowToPreserveStackDetails.Fixer.cs

```diff
@@ -0,0 +1,41 @@
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+
+namespace Microsoft.CodeQuality.Analyzers.QualityGuidelines
+{
+    /// <summary>
+    /// CA2200: Rethrow to preserve stack details
+    /// </summary>
+    public abstract class RethrowToPreserveStackDetailsFixer : CodeFixProvider
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(RethrowToPreserveStackDetailsAnalyzer.RuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers'
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+        public async sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var diagnostic = context.Diagnostics.First();
+            var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+            var nodeToReplace = root.FindNode(diagnosticSpan);
```

Comment:
Couple of comments:

1. You don't need to get the first diagnostic or use it's span
2. Please add a null check for nodeToReplace

```csharp
            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
            var nodeToReplace = root.FindNode(context.Span);
            if (nodeToReplace == null)
            {
                return;
            }

            context.RegisterCodeFix(
                CodeAction.Create(...),
                context.Diagnostics);
```

Reply:
Uh, Ok, I just got it from the boilerplate https://docs.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/tutorials/how-to-write-csharp-analyzer-code-fix  and it does the job as well. I will modify it.